docs/pull_request.md

```diff
@@ -0,0 +1,208 @@
+
+It's also applicable for code rework, or code formatting. Sometimes, it is more efficient to extract that work to a dedicated PR, and rebase your branch once this "rework" PR has been merged.
+
+#### Bots
+
+Some bots can create PR, but they still have to be reviewed by the team
+
+##### Dependabot
+
+Dependabot is a tool which maintain all our external dependencies up to date. A dedicated PR is created for each new available release for one of our external dependency.Dependabot
+
+To review such PR, you have to
+ - **IMPORTANT** check the diff files (as always).
+ - Check the release note. Some existing bugs in Element project may be fixed by the upgrade
+ - Make sure that the CI is happy
+ - If the code does not compile (API break for instance), you have to checkout the branch and push new commits
+ - Do some smoke test, depending of the library which has been upgraded
+
+For some reason dependabot sometimes does not upgrade some dependencies. In this case, and when detected, the upgrade has to be done manually.
+
+##### Gradle wrapper
+
+`Update Gradle Wrapper` is a tool which can create PR to upgrade our gradle.properties file.
+Review such PR is the same recipe than for PR from Dependabot
+
+##### Sync analytics plan
+
+This tools imports any update in the analytics plan. See instruction in the PR itself to handle it.
+More info can be found in the file [analytics.md]
+
+## Reviewing PR
+
+### Who can review pull requests?
+
+As an open source project, every one can review each others PR. Of course an approval from internal developer is mandatory for a PR to be merged.
+But comment in PR from the community are always appreciated!
+
+### What to have in mind when reviewing a PR
+
+1. User experience: is the UX and UI correct? You will probably be the second person to test the new thing, the first one is the developer.
+2. Developer experience: does the code look nice and decoupled? No big functions, new classes added to the right module, etc.
+3. Code maintenance. A bit similar to point 2. Tricky code must be documented for instance
+4. Fork consideration. Will configuration of forks be easy? Some documentation may help in some cases.
+5. We are building long term products. "Quick and dirty" code must be avoided.
+6. The PR includes new tests for the added code, updated test for the existing code
+7. All PRs from external contributors **MUST** include a sign-off. It's in the checklist, and sometimes it's checked by the submitter, but there is actually no sign-off. In this case, ask nicely for a sign-off and request changes (do not approve the PR, even if everything else is fine).
+
+### Rules
+
+#### Check the form
+
+##### PR description
+
+PR description should follow the PR template, and at least provide some context about the code change.
+
+##### File change
+
+1. Code should follow the guidelines
+2. Code should be formatted correctly
+3. XML attribute must be sorted
```

Comment:
will the CI - lint/code quality script catch these? we might be able to delegate these points to the guidelines docs  (would be handy if the guidelines was a link)